ethers-core/src/utils/mod.rs

```diff
@@ -155,19 +155,28 @@ where
     S: ToString,
     K: TryInto<Units, Error = ConversionError> + Copy,
 {
-    use rust_decimal::{Decimal, MathematicalOps};
-
-    let num: Decimal = amount.to_string().parse()?;
-    let exponent = units.try_into()?.as_num();
-
-    let multiplier = Decimal::TEN
-        .checked_powu(exponent.into())
-        .ok_or(rust_decimal::Error::ExceedsMaximumPossibleValue)?;
+    let exponent: u32 = units.try_into()?.as_num();
+    let mut a_str = amount.to_string().replace("_", "");
+    let dot_i = if let Some(di) = a_str.find('.') {
+        a_str.remove(di);
+        di
+    } else {
+        a_str.len()
+    };
+    let a_len = a_str.len();
+    let dec_len = (a_len - dot_i) as u32;
 
-    let val =
-        num.checked_mul(multiplier).ok_or(rust_decimal::Error::ExceedsMaximumPossibleValue)?;
-    let u256_n: U256 = U256::from_dec_str(&val.round().to_string())?;
-    Ok(u256_n)
+    if dec_len > exponent {
+        let a_str = &a_str[..(a_len - (dec_len - exponent) as usize)];
```

Comment:
what's the rational behind this?

let's document this

Reply:
Does "Truncate the decimal part if it is longer than the exponent" describe it well enough?